src/lib/template.ts

```diff
@@ -41,11 +41,32 @@ export const boundAttributeSuffix = '$lit$';
  * An updateable Template that tracks the location of dynamic parts.
  */
 export class Template {
-  parts: TemplatePart[] = [];
-  element: HTMLTemplateElement;
+  private _parts: TemplatePart[]|undefined = undefined;
+  private _element: HTMLTemplateElement|undefined = undefined;
+  result: TemplateResult;
```

Comment:
I don't like that a template is holding on to a result that created it. TemplateResults should be fairly transient, and this will also hold onto the values a Template was first rendered with.